### ISSUE TITLE
Minification of pre-js file in 1.38.27 with WASM=0

### ISSUE THREAD

**Issue by BenTalagan** [emscripten-core/emscripten#8072]
Hi, I'm not totally sure of what's happening here. I have a project on which I use the file packager to create a package which is passed to em++ as a --pre-js file. In the final output, when using WASM=1 everything is minified. But when passing WASM=0, the package part of the final compiled file is not minified. It used to work with 1.38.25 but after an upgrade 1.38.27 that behaviour has changed (resulting in an overhead of 1MB over 3MB when not using WASM). Here is my complete list of compilation flags : 

-O3 -s RESERVED_FUNCTION_POINTERS=2 --memory-init-file 0 -s SINGLE_FILE -s FORCE_FILESYSTEM=1 -s WASM=0

Thanks :)

**Comment by juj** [comment]
Bisected this to have regressed in

```
59f08fb3ab045a046d89646a4df8d2a6050c3161 is the first bad commit
commit 59f08fb3ab045a046d89646a4df8d2a6050c3161
Author: Alon Zakai <alonzakai@gmail.com>
Date:   Fri Feb 8 13:59:52 2019 -0800

    Use ESTree-based optimizer for core JS passes, to support ES6+ inputs (#7973)
    
    Fixes #6000
    
    The key change here is to rewrite the JS optimizer passes that run in a normal `-O3` etc. build from the Uglify1 AST to ESTree. With ESTree we can use modern parsers etc. so that we support ES6+ inputs to js libraries, pre-jses, EM_ASM, etc.
    
    Aside from that rewrite, the other changes are less critical and can be altered later. Specifically, this uses acorn for parsing and terser for outputting, but we could switch to anything using ESTree very easily. Acorn is nice for parsing since it's small and standalone. For outputting I experimented with astring, which is small and nice, and escodegen, which looks very robust, but neither could output compact-enough JS to not regress our JS code sizes. This is not truly critical since for minimal code size people should use closure anyhow, however, it's nice for default builds to be small (and we don't run closure by default), and I didn't want to regress anything. Using the terser outputter achieves that. (Since it uses the Uglify2 AST internally, this means using their tool to convert ESTree to Uglify2.) They may be some minor code size changes with this PR, just because we use a different outputter now, but nothing major in either direction. Most changes seem positive actually. Sizes after closure are unchanged.
    
    This uses almost unmodified versions of acorn and terser, but they are stripped down to what we need, and I had to make two modifications, see these PRs: [acornjs/acorn#793](https://github.com/acornjs/acorn/pull/793) (quote the error on parse exceptions) and [mishoo/UglifyJS2#3323](https://github.com/mishoo/UglifyJS2/pull/3323) (preserve quoted properties).
    
    This may very slightly regress compile times when using those passes, as Uglify1 was just very fast. However, the change should be very small.
    
    This does _not_ rewrite every single JS optimizer pass. In particular the asm.js passes don't need to support ES6, and so don't need to be rewritten. There are also optional passes that do not run by default, that we can convert later depending on priority.

:040000 040000 b0ec09ce42fa9aa228f796ec2e87cea8d3ad9414 f590e73a2b1f26570f423ecfabceced225b75d4c M      tests
:040000 040000 3f8e4d7ca451323488bd1777ba4a3cdf71cced7d 6c96a1b585fe22696baa1ab5a055b39d9b57237b M      tools
```

A short repro is
```
emcc -O3 -s WASM=0 tests/hello_world.c -o a.js && tail -n 20 a.js
```

**Comment by BenTalagan** [comment]
Thanks for having had a look at it! Wasn't even sure if I was dreaming, and would have been really sorry for the noise.

**Comment by kripken** [comment]
Thanks for the bugreport and for the bisection! Fix in #8074, sorry about this regression.
